### ISSUE TITLE
iguodalagladys-ship-it/insurance-claim-prediction

### ISSUE THREAD

**Issue by iguodalagladys-ship-it** [gbganalyst/Machine-Learning-Class#49]

**Comment by iguodalagladys-ship-it** [comment]
My capstone project 
Project Title:


Insurance Claim Prediction Based on Building Characteristics


 Project Objective

The goal of this project is to build a predictive classification model that estimates the probability that a building will make at least one insurance claim during its insured period.

Target variable: Claim

1 → Building has at least one claim

0 → No claim

The model is built using building characteristics and evaluated using appropriate classification metrics

.

 GitHub Repository Structure 


insurance-claim-prediction/

│

├── data/

│   ├── Train_data.csv

│   └── Variable Description.csv

│

├── notebooks/

│   └── insurance_claim_prediction.ipynb

│

├── README.md

└── requirements.txt


Notebook Content 

 1. Import Libraries


Python

import pandas as pd

import numpy as np

import matplotlib.pyplot as plt

import seaborn as sns


from sklearn.model_selection import train_test_split

from sklearn.preprocessing import StandardScaler, OneHotEncoder

from sklearn.compose import ColumnTransformer

from sklearn.pipeline import Pipeline


from sklearn.linear_model import LogisticRegression

from sklearn.ensemble import RandomForestClassifier

from sklearn.metrics import (

    accuracy_score, 

    precision_score, 

    recall_score, 

    f1_score, 

    roc_auc_score, 

    classification_report, 

    confusion_matrix

)


import warnings

warnings.filterwarnings('ignore')

 2. Load the Dataset


Python

df = pd.read_csv("../data/Train_data.csv")

df.head()

 3. Data Understanding


Python

df.shape

df.info()

df.describe()

Explain in markdown:

Number of rows & columns

Data types

Presence of categorical and numerical variables

4. Target Variable Distribution


Python

sns.countplot(x='Claim', data=df)

plt.title("Distribution of Insurance Claims")

plt.show()


df['Claim'].value_counts(normalize=True)

Insight example:

The dataset is imbalanced, with fewer buildings making claims. This justifies the use of ROC-AUC and F1-score for evaluation.


Data Cleaning & Preprocessing

Missing Values


Python

df.isnull().sum().sort_values(ascending=False)

Handle missing values:

Numerical → median

Categorical → mode

Copy code

Python

num_cols = df.select_dtypes(include=np.number).columns

cat_cols = df.select_dtypes(include='object').columns


for col in num_cols:

    df[col].fillna(df[col].median(), inplace=True)


for col in cat_cols:

    df[col].fillna(df[col].mode()[0], inplace=True)


 Outlier Check 


Python

sns.boxplot(data=df[num_cols])

plt.xticks(rotation=90)

plt.show()


Exploratory Data Analysis (EDA)

 Numerical Features vs Claim


Python

for col in num_cols:

    if col != 'Claim':

        sns.boxplot(x='Claim', y=col, data=df)

        plt.title(f"{col} vs Claim")

        plt.show()


 Categorical Features vs Claim


Python

for col in cat_cols:

    sns.countplot(x=col, hue='Claim', data=df)

    plt.xticks(rotation=45)

    plt.title(f"{col} vs Claim")

    plt.show()


 Feature Engineering & Modelling Preparation

Separate Features and Target


Python

X = df.drop('Claim', axis=1)

y = df['Claim']

Train-Test Split

Copy code

Python

X_train, X_test, y_train, y_test = train_test_split(

    X, y, test_size=0.2, random_state=42, stratify=y

)

Column Transformer


Python

numeric_features = X.select_dtypes(include=np.number).columns

categorical_features = X.select_dtypes(include='object').columns


preprocessor = ColumnTransformer(

    transformers=[

        ('num', StandardScaler(), numeric_features),

        ('cat', OneHotEncoder(handle_unknown='ignore'), categorical_features)

    ]

)

 Model 1: Logistic Regression


Python

log_model = Pipeline(steps=[

    ('preprocessor', preprocessor),

    ('classifier', LogisticRegression(max_iter=1000))

])


log_model.fit(X_train, y_train)

y_pred_log = log_model.predict(X_test)

y_prob_log = log_model.predict_proba(X_test)[:,1]


Evaluation


Python

print("Logistic Regression Performance")

print(classification_report(y_test, y_pred_log))

print("ROC-AUC:", roc_auc_score(y_test, y_prob_log))


 Model 2: Random Forest


Python

rf_model = Pipeline(steps=[

    ('preprocessor', preprocessor),

    ('classifier', RandomForestClassifier(

        n_estimators=200,

        random_state=42,

        class_weight='balanced'

    ))

])


rf_model.fit(X_train, y_train)

y_pred_rf = rf_model.predict(X_test)

y_prob_rf = rf_model.predict_proba(X_test)[:,1]


Evaluation


Python

print("Random Forest Performance")

print(classification_report(y_test, y_pred_rf))

print("ROC-AUC:", roc_auc_score(y_test, y_prob_rf))


Model Comparison Summary (Markdown)

Model

Accuracy

Precision

Recall

F1

ROC-AUC

Logistic Regression

…

…

…

…

…

Random Forest

…

…

…

…

 Conclusion & Business Insight

Example:

The Random Forest model performed better in capturing claim-prone buildings, achieving a higher ROC-AUC score. This model can assist insurers in risk-based pricing and proactive risk management.


1️⃣1️⃣ README.md 

Include:

Project overview

Dataset description

Models used

Evaluation metrics

How to run the notebook


1️⃣2️⃣ Push to GitHub


Bash

git init

git add .

git commit -m "Insurance claim prediction project"

git branch -M main

git remote add origin https://github.com/yourusername/insurance-claim-prediction.git

git push -u origin main

**Comment by gbganalyst** [comment]
I did not see any capstone project. Do you want to check any?

**Comment by iguodalagladys-ship-it** [comment]
Project Title:


Insurance Claim Prediction Based on Building Characteristics


 Project Objective

The goal of this project is to build a predictive classification model that estimates the probability that a building will make at least one insurance claim during its insured period.

Target variable: Claim

1 → Building has at least one claim

0 → No claim

The model is built using building characteristics and evaluated using appropriate classification metrics

.

 GitHub Repository Structure 


insurance-claim-prediction/

│

├── data/

│   ├── Train_data.csv

│   └── Variable Description.csv

│

├── notebooks/

│   └── insurance_claim_prediction.ipynb

│

├── README.md

└── requirements.txt


Notebook Content 

 1. Import Libraries


Python

import pandas as pd

import numpy as np

import matplotlib.pyplot as plt

import seaborn as sns


from sklearn.model_selection import train_test_split

from sklearn.preprocessing import StandardScaler, OneHotEncoder

from sklearn.compose import ColumnTransformer

from sklearn.pipeline import Pipeline


from sklearn.linear_model import LogisticRegression

from sklearn.ensemble import RandomForestClassifier

from sklearn.metrics import (

    accuracy_score, 

    precision_score, 

    recall_score, 

    f1_score, 

    roc_auc_score, 

    classification_report, 

    confusion_matrix

)


import warnings

warnings.filterwarnings('ignore')

 2. Load the Dataset


Python

df = pd.read_csv("../data/Train_data.csv")

df.head()

 3. Data Understanding


Python

df.shape

df.info()

df.describe()

Explain in markdown:

Number of rows & columns

Data types

Presence of categorical and numerical variables

4. Target Variable Distribution


Python

sns.countplot(x='Claim', data=df)

plt.title("Distribution of Insurance Claims")

plt.show()


df['Claim'].value_counts(normalize=True)

Insight example:

The dataset is imbalanced, with fewer buildings making claims. This justifies the use of ROC-AUC and F1-score for evaluation.


Data Cleaning & Preprocessing

Missing Values


Python

df.isnull().sum().sort_values(ascending=False)

Handle missing values:

Numerical → median

Categorical → mode

Copy code

Python

num_cols = df.select_dtypes(include=np.number).columns

cat_cols = df.select_dtypes(include='object').columns


for col in num_cols:

    df[col].fillna(df[col].median(), inplace=True)


for col in cat_cols:

    df[col].fillna(df[col].mode()[0], inplace=True)


 Outlier Check 


Python

sns.boxplot(data=df[num_cols])

plt.xticks(rotation=90)

plt.show()


Exploratory Data Analysis (EDA)

 Numerical Features vs Claim


Python

for col in num_cols:

    if col != 'Claim':

        sns.boxplot(x='Claim', y=col, data=df)

        plt.title(f"{col} vs Claim")

        plt.show()


 Categorical Features vs Claim


Python

for col in cat_cols:

    sns.countplot(x=col, hue='Claim', data=df)

    plt.xticks(rotation=45)

    plt.title(f"{col} vs Claim")

    plt.show()


 Feature Engineering & Modelling Preparation

Separate Features and Target


Python

X = df.drop('Claim', axis=1)

y = df['Claim']

Train-Test Split

Copy code

Python

X_train, X_test, y_train, y_test = train_test_split(

    X, y, test_size=0.2, random_state=42, stratify=y

)

Column Transformer


Python

numeric_features = X.select_dtypes(include=np.number).columns

categorical_features = X.select_dtypes(include='object').columns


preprocessor = ColumnTransformer(

    transformers=[

        ('num', StandardScaler(), numeric_features),

        ('cat', OneHotEncoder(handle_unknown='ignore'), categorical_features)

    ]

)

 Model 1: Logistic Regression


Python

log_model = Pipeline(steps=[

    ('preprocessor', preprocessor),

    ('classifier', LogisticRegression(max_iter=1000))

])


log_model.fit(X_train, y_train)

y_pred_log = log_model.predict(X_test)

y_prob_log = log_model.predict_proba(X_test)[:,1]


Evaluation


Python

print("Logistic Regression Performance")

print(classification_report(y_test, y_pred_log))

print("ROC-AUC:", roc_auc_score(y_test, y_prob_log))


 Model 2: Random Forest


Python

rf_model = Pipeline(steps=[

    ('preprocessor', preprocessor),

    ('classifier', RandomForestClassifier(

        n_estimators=200,

        random_state=42,

        class_weight='balanced'

    ))

])


rf_model.fit(X_train, y_train)

y_pred_rf = rf_model.predict(X_test)

y_prob_rf = rf_model.predict_proba(X_test)[:,1]


Evaluation


Python

print("Random Forest Performance")

print(classification_report(y_test, y_pred_rf))

print("ROC-AUC:", roc_auc_score(y_test, y_prob_rf))


Model Comparison Summary (Markdown)

Model

Accuracy

Precision

Recall

F1

ROC-AUC

Logistic Regression

…

…

…

…

…

Random Forest

…

…

…

…

 Conclusion & Business Insight

Example:

The Random Forest model performed better in capturing claim-prone buildings, achieving a higher ROC-AUC score. This model can assist insurers in risk-based pricing and proactive risk management.


1️⃣1️⃣ README.md 

Include:

Project overview

Dataset description

Models used

Evaluation metrics

How to run the notebook


1️⃣2️⃣ Push to GitHub


Bash

git init

git add .

git commit -m "Insurance claim prediction project"

git branch -M main

git remote add origin https://github.com/yourusername/insurance-claim-prediction.git

git push -u origin main
